E0318 needs to be updated to new format

From: src/test/compile-fail/typeck-default-trait-impl-outside-crate.rs

Error E0318 needs a span label, a note, and an updated title, updating it from:

```
error[E0318]: cannot create default implementations for traits outside the crate they're defined in; define a new trait instead
  --> src/test/compile-fail/typeck-default-trait-impl-outside-crate.rs:13:1
   |
13 | impl Copy for .. {}
   | ^^^^^^^^^^^^^^^^^^^
```

To:

```
error[E0318]: cannot create default implementations for traits outside the crate they're defined in
  --> src/test/compile-fail/typeck-default-trait-impl-outside-crate.rs:13:1
   |
13 | impl Copy for .. {}
   | ^^^^^^^^^^^^^^^^^^^ `Copy` trait not defined in this crate
   |
   = note: define a new trait instead 
```

Bonus: narrow the span to focus on the foreign trait

```
error[E0318]: cannot create default implementations for traits outside the crate they're defined in
  --> src/test/compile-fail/typeck-default-trait-impl-outside-crate.rs:13:1
   |
13 | impl Copy for .. {}
   |      ^^^^ trait not defined in this crate
   |
   = note: define a new trait instead 
```

--- a/src/librustc_typeck/coherence/orphan.rs
+++ b/src/librustc_typeck/coherence/orphan.rs
@@ -347,15 +347,19 @@ impl<'cx, 'tcx> OrphanChecker<'cx, 'tcx> {
                     return;
                 }
             }
-            hir::ItemDefaultImpl(..) => {
+            hir::ItemDefaultImpl(_, ref item_trait_ref) => {
                 // "Trait" impl
                 debug!("coherence2::orphan check: default trait impl {}",
                        self.tcx.map.node_to_string(item.id));
                 let trait_ref = self.tcx.impl_trait_ref(def_id).unwrap();
                 if trait_ref.def_id.krate != LOCAL_CRATE {
-                    span_err!(self.tcx.sess, item.span, E0318,
+                    struct_span_err!(self.tcx.sess, item_trait_ref.path.span, E0318,
                               "cannot create default implementations for traits outside the \
-                               crate they're defined in; define a new trait instead");
+                               crate they're defined in; define a new trait instead")
+                        .span_label(item_trait_ref.path.span,
+                                    &format!("`{}` trait not defined in this crate",
+                                             item_trait_ref.path))
+                        .emit();
                     return;
                 }
             }

--- a/src/test/compile-fail/typeck-default-trait-impl-outside-crate.rs
+++ b/src/test/compile-fail/typeck-default-trait-impl-outside-crate.rs
@@ -10,7 +10,6 @@
 
 #![feature(optin_builtin_traits)]
 
-impl Copy for .. {}
-//~^ ERROR E0318
-
+impl Copy for .. {} //~ ERROR E0318
+     //~^ NOTE `Copy` trait not defined in this crate
 fn main() {}

